pandas-stubs/core/indexes/range.pyi

```diff
@@ -16,18 +16,9 @@ from pandas._typing import (
 )
 
 class RangeIndex(_IntIndexType):
-    def __new__(
-        cls,
-        start: int | RangeIndex = ...,
-        stop: int = ...,
-        step: int = ...,
-        dtype=...,
-        copy: bool = ...,
-        name=...,
-    ): ...
     def __init__(
-        self,
-        start: int | RangeIndex = ...,
+        cls,
```

Comment:
This should be `self`, because it is in `__init__()`

I think this is the cause of the pyright error you are getting.